dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/externaltools/ExternalToolServiceBean.java

```diff
@@ -98,17 +107,39 @@ public ExternalTool save(ExternalTool externalTool) {
         return em.merge(externalTool);
     }
 
+    // Used only in REST
+    public static List<ExternalTool> findExternalToolsByFile(List<ExternalTool> allExternalTools, DataFile file) {
+        // Map tabular data to it's mimetype (the isTabularData() check assures that this code works the same as before,
+        // but it may need to change if tabular data is split into subtypes with differing mimetypes)
+        final String contentType = file.isTabularData() ? TextMimeType.TSV_ALT.getMimeValue() : file.getContentType();
+
+        // Match tool and file type
+        return allExternalTools.stream()
+                .filter(t -> t.getContentType().equals(contentType))
+                .collect(Collectors.toList());
+    }
+
     /**
-     * This method takes a list of tools and a file and returns which tools that file supports
-     * The list of tools is passed in so it doesn't hit the database each time
+     * This method takes a list of tools, a file and a dataset version and
+     * returns which tools that file supports. The list of tools is passed in
+     * so it doesn't hit the database each time
      */
-    public static List<ExternalTool> findExternalToolsByFile(List<ExternalTool> allExternalTools, DataFile file) {
+    public List<ExternalTool> findExternalToolsByFileAndVersion(
+            List<ExternalTool> allExternalTools, DataFile file, DatasetVersion datasetVersion) {
         List<ExternalTool> externalTools = new ArrayList<>();
-        //Map tabular data to it's mimetype (the isTabularData() check assures that this code works the same as before, but it may need to change if tabular data is split into subtypes with differing mimetypes)
+        // Map tabular data to it's mimetype (the isTabularData() check assures that this code works the same as before,
+        // but it may need to change if tabular data is split into subtypes with differing mimetypes)
         final String contentType = file.isTabularData() ? TextMimeType.TSV_ALT.getMimeValue() : file.getContentType();
+        boolean isTsvAltContentType = contentType.equals(TextMimeType.TSV_ALT.getMimeValue());
+
         allExternalTools.forEach((externalTool) -> {
-            //Match tool and file type 
+            // Match tool and file type
             if (contentType.equals(externalTool.getContentType())) {
+
+                // Exclude non-public tsv files and those which do not have datatable
+                if (isTsvAltContentType && (!isFilePublic(file, datasetVersion) || !file.isTabularData())) {
+                    return;
+                }
```

Comment:
Can we please avoid using comments and use private methods with descriptive names instead?
Here for example something like `if (isTsvNonPublicOrNonTabular) ` or something like that.